web/src/main/ui/app/components/flows-new/edit-flow/ui/new-step-dialog-ui.component.ts

```diff
@@ -22,12 +19,13 @@ export class NewStepDialogUiComponent implements OnInit {
   @Input() collections: any;
   @Input() step: any;
   @Input() flow: Flow;
+  @Input() projectDirectory: string;
   @Output() getCollections = new EventEmitter();
   @Output() cancelClicked = new EventEmitter();
   @Output() saveClicked = new EventEmitter();
 
-  public newStep: Step = new Step;
-  readonly stepOptions = ['ingest', 'mapping', 'mastering', 'custom'];
+  public newStep: Step;
+  readonly stepOptions = ['INGESTION', 'MAPPING', 'MASTERING', 'CUSTOM'];
```

Comment:
Because there is a number of other files that use step types as a string, it becomes dangerous to do refactorings like these all the time. I suggest also along the Step model type add an enum like this:
``` typescript
export enum StepType {
  INGESTION = "INGESTION",
  MAPPING = "MAPPING",
  MASTERING = "MASTERING",
  CUSTOM = "CUSTOM"
};

export class Step {
  public id: string;
  public type: StepType;
}
```
and update all string references used in different expressions in templates as well as in TS files.

Reply:
Agreed. Backend does this for the exact reason, we also will be adding new types for sure. So this is an excellent suggestion by @alexsmr